java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowFileWriter.java

```diff
@@ -69,6 +70,31 @@ public ArrowFileWriter(VectorSchemaRoot root, DictionaryProvider provider, Writa
     this.metaData = metaData;
   }
 
+  public ArrowFileWriter(VectorSchemaRoot root, boolean includeNullCount, CompressionCodec codec,
```

Comment:
Instead of making an ever expanding list of constructors, I think it is likely better at this point to create an "Options"  class that can be built using a builder pattern and passed in.  So we can have one new constructor:

ArrowFileWriter(VectorSchemaRoot root, WriteOptions options, WritableByteChannel) and then maybe one private constructor if necessary that explodes the options.